5/21 Release notes (#3764)
astronomer/docs: astro/release-notes.md

<!-- ALL LINKS TO INTERNAL DOCS MUST BE COMPLETE URLS INCLUDING HTTPS. Otherwise the links will break in RSS. -->

> ## May 21, 2024

Author: ```suggestion
## May 22, 2024
```

> ### Official Astro GitHub integration is now in Public Preview

Reviewer: ```suggestion
### The Astro GitHub integration is now in Public Preview
```

:::publicpreview
:::

The ability to deploy automatically from GitHub using the official Astro GitHub integration is now in Public Preview.

The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro by merging pull requests or making commits directly to specific branches, without needing to configure a GitHub Action. Additionally, the GitHub integration displays Git metadata directly in the Astro UI, including Git commit descriptions and gives you greater visibility into the status and logs of individual code deploys.

See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.

### Bug fixes

> - Fixed a bug where users using custom roles in the UI could see roles that could not be assigned.

Reviewer: ```suggestion
- Fixed an issue where users with custom roles could see roles that could not be assigned in the Astro UI.
```
Where are they seeing the roles? Is this users that have custom roles, or users who are creating/ assigning custom roles? Some clarifying details here would be helpful

Author: The clarifying details you added are my understanding of the bug fix!

> - Fixed a bug where users creating custom roles in the UI could see roles that could not be assigned. <!-- https://github.com/astronomer/astro/pull/21148-->

Reviewer 2: "users creating custom roles" should say "users using custom roles"

> - Fixed a bug where users creating custom roles in the UI could see roles that could not be assigned. <!-- https://github.com/astronomer/astro/pull/21148-->

Author: ```suggestion
- Fixed a bug where users using custom roles in the UI could see roles that could not be assigned.
```
